### ISSUE TITLE
false positive for E301

### ISSUE THREAD

**Issue by wookietreiber** [ansible-community/ansible-lint#490]
# Issue Type
- Bug report

# Ansible and Ansible Lint details

```console
$ ansible --version
ansible 2.7.8
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.7.2 (default, Jan 10 2019, 23:51:51) [GCC 8.2.1 20181127]

$ ansible-lint --version
ansible-lint 4.1.0
```

- ansible installation method: OS package (Arch Linux)
- ansible-lint installation method: OS package (Arch Linux)

# Desired Behavior

There should be no false positive output. There could also be a directive to explicitly tell **ansible-lint** where to look for the `when`, see example below.

# Example 1: include_tasks

`pb.yml`:

```yml
---

- name: test playbook
  hosts: localhost
  roles:
    - test
  vars:
    foo: yes

...
```

`roles/test/tasks/main.yml`:

```yml
---

# note: foo comes from vars in this example
#       in practice, it comes from other command that compares installed
#       version with version from vars
- name: include update foo tasks
  include_tasks: foo.yml
  when: foo

...
```

`roles/test/tasks/foo.yml`:

```yml
---

# note: can't use creates here because we update something that might
#       already be there
# note: I don't need to use when here because when is already used with
#       include_tasks
# note: the next line is the example for a possible ansible-lint directive
#       letting ansible-lint know where to look for the when
# ansible-lint#301: when-included
- name: update foo
  command: update-foo

...
```

# Example 2: block

`pb.yml`:

```yml
---

- name: test playbook
  hosts: localhost
  roles:
    - test
  vars:
    foo: yes

...
```

`roles/test/tasks/main.yml`:

```yml
---

# note: foo comes from vars in this example
#       in practice, it comes from other command that compares installed
#       version with version from vars
- name: include update foo tasks
  when: foo
  block:

    # note: can't use creates here because we update something that might
    #       already be there
    # note: I don't need to use when here because when is already used with
    #       include_tasks
    - name: update foo
      command: update-foo

...
```

# Actual Behavior

The check E301 is not aware of the `when` that is used at `include_tasks`, it only seems to check for a `when` directly on the `command` task. The `block` construct gives no false positive.

```console
# example 1
$ ansible-lint pb.yml
[301] Commands should not change things if nothing needs doing
/home/umcdev/projects/ansible/roles/foo/roles/test/tasks/foo.yml:7
Task/Handler: update foo

# example 2 -- this works already
$ ansible-lint pb.yml
$ echo $?
0
```

**Comment by hwwilliams** [comment]
Experiencing this same issue with a couple of my rules that have tasks other than main that are conditional task files.

**Comment by 7php** [comment]
hi any updates on this?
It's annoying to have false positives when linting.

**Comment by utrumo** [comment]
I have problem with this rule too:
`roles/virtualbox-guest/tasks/remove.yml:15: [E301] [HIGH] Commands should not change things if nothing needs doing`
If i run this task - it removes /opt/VboxGuestAdditional* folder and in the next launch it will do nothing.
What i'm do wrong?
```
- name: Remove guest tools
  command: /opt/VBoxGuestAdditions-{{ guest_version }}/uninstall.sh
  removes: /opt/VBoxGuestAdditions-{{ guest_version }}/
```

**Comment by ssbarnea** [comment]
When using `command` or `shell` you must also mention `changed_when` in order to help ansible figure-out if a command modified something or not. In the end Ansible has no clue about what you executed, so it cannot make an informed decision.

You are welcomed to propose a PR to improve the linter message as it seems to confuse enough people.

**Comment by ssbarnea** [comment]
@webknjaz I would close it, as a false-positive bug report ;)

**Comment by utrumo** [comment]
EE

> When using `command` or `shell` you must also mention `changed_when` in order to help ansible figure-out if a command modified something or not. In the end Ansible has no clue about what you executed, so it cannot make an informed decision.
> 
> You are welcomed to propose a PR to improve the linter message as it seems to confuse enough people.

Interestingly. I thought the removed or created directive would suffice...
